### PR TITLE
Thomas/sorted pt

### PR DESCRIPTION
**Pull request by Mostah**

### CODE REVIEW

File: src/Performance.h [hunk 1]
```diff
@@ -65,6 +65,14 @@ class Performance {
...
 
+  /**
+   * getCriteria get of the criteria name used in the performance
+   *
+   * @return crit_vect string vector containing the criterion names used in this
+   * performance.
+   */
+  std::vector<std::string> getCriteria();
+
   /**
```
Comment (thibmonsel) [inline src/Performance.h]: change the method function to get CriteriaId() to remove ambiguity

File: src/PerformanceTable.cpp [hunk 2]
```diff
@@ -1,49 +1,230 @@
...
+    std::sort(pv.begin(), pv.end(), [](const Perf &a, const Perf &b) {
+      return a.getValue() < b.getValue();
+    });
+  }
+}
+
+std::vector<Perf> PerformanceTable::getAltBetween(std::string crit, float inf,
+                                                  float sup) {
+  if (mode_ != "crit") {
+    throw std::domain_error("Mode must be crit but is currently set to " +
```
Comment (thibmonsel) [inline src/PerformanceTable.cpp:212]: Lets call the argument std::string with a more meaningful name ie are we talking about criterion, criteria and what ? (ie criterion id, criterion name etc ...)

**Review by thibmonsel — APPROVED** [review]

File: src/PerformanceTable.cpp [hunk 3]
```diff
@@ -1,49 +1,306 @@
...
+void PerformanceTable::generateRandomPerfValues(unsigned long int seed) {
+  srand(seed);
+  for (std::vector<Perf> &pv : pt_) {
+    for (Perf &p : pv) {
+      p.setValue((float)rand() / RAND_MAX);
+    }
+  }
+  sorted_ = false;
+}
+
```
Comment (thibmonsel) [inline src/PerformanceTable.cpp]: should we use a have a lower and upper bound limit for the values specified ? 
ie by default its between 0 and 1 but you can give btwn 1 and 4 for example ?

**Review by thibmonsel — APPROVED** [review]

File: src/PerformanceTable.cpp (hunk 3 above)
Comment (Mostah) [reply src/PerformanceTable.cpp]: Yeah we could add this, but aren't the values supposed to be in [0,1] ?

Comment (thibmonsel) [reply src/PerformanceTable.cpp]: I think they can be as big as one want but for optimization purpose the data probably needs to be normalised

Comment (Mostah) [reply src/PerformanceTable.cpp]: I see, we will need to add these normalization functions! I will add some bound limit args here to be more flexible

File: src/PerformanceTable.h [hunk 4]
```diff
@@ -19,10 +21,11 @@ class PerformanceTable {
    * Criteria constructor without perf values but set of criteria to evaluate
```
Comment (edix9) [inline src/PerformanceTable.h]: Performances instead of Criteria?

**Pull request merged**